src/hooks/useAsyncSwap.ts

```diff
@@ -0,0 +1,92 @@
+import { chainIdAtom, walletAtom } from '@/state/atoms'
+import { RESERVE_API } from '@/utils/constants'
+import { asyncSwapEndpointAtom } from '@/views/index-dtf/issuance/async-swaps/atom'
+import { AsyncSwapResponse } from '@/views/index-dtf/issuance/async-swaps/types'
+import { useQuery } from '@tanstack/react-query'
+import { useAtomValue, useSetAtom } from 'jotai'
+import mixpanel from 'mixpanel-browser/src/loaders/loader-module-core'
+import { useEffect, useMemo } from 'react'
+import { Address } from 'viem'
+import useDebounce from './useDebounce'
+
+const useAsyncSwap = ({
+  dtf,
+  amountOut,
+  slippage,
+  disabled,
+  dtfTicker,
+  type,
+}: {
+  dtf?: Address
+  amountOut: string
+  slippage: number
+  disabled: boolean
+  dtfTicker: string
+  type: 'mint' | 'redeem'
+}) => {
+  const chainId = useAtomValue(chainIdAtom)
+  const account = useAtomValue(walletAtom)
+  const setAsyncSwapEndpoint = useSetAtom(asyncSwapEndpointAtom)
+
+  const getEndpoint = () =>
+    !dtf || isNaN(Number(amountOut)) || Number(amountOut) === 0
+      ? null
+      : `${RESERVE_API}async-swap/quote?dtf=${dtf}&chainId=${chainId}&amountOut=${amountOut}&operation=${type}&signer=${account}`
+
+  const endpoint = useDebounce(
+    useMemo(
+      () => getEndpoint(),
+      [chainId, account, dtf, amountOut, type, slippage]
+    ),
+    500
+  )
+
+  useEffect(() => {
+    setAsyncSwapEndpoint(endpoint ?? '')
```

Comment:
why? is only used here, and if its going to be used on other module, then coupling it in this hook would be weird as this hooks is serving an specific purpose

Reply:
I'm going to remove it. I was actually leaving it because in Zaps we use it to send to Mixpanel as the event ID (and to extract all the sent parameters from there). I figured we'd need it here too, but for now I'm removing it.